### PR TITLE
Simple architecture

### PR DESCRIPTION
**Pull request by Swirastlynn**
Screen with examples:
1. Simple list with asynchronous data load
2. Provider-Composer based architecture for a state in a piechart

### CODE REVIEW

**Comment by lenkrzysztof** [comment]
Which version of Flutter do you use? If 2.5 your constants value should have const keyword before like: `const Text('example');`. You should checked your terminal due to warning then.

File: lib/people/people_list_screen.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.white,
+      appBar: AppBar(
+        title: Text('People List'),
+      ),
+      body: ListView.builder(
+        itemCount: widgets.length,
+        itemBuilder: (BuildContext context, int position) {
```
Comment (lenkrzysztof) [inline lib/people/people_list_screen.dart]: You can omit unused parameters with underscore: 
`itemBuilder: (_, int position) {`

File: lib/people/people_list_screen.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+class PeopleListScreen extends StatefulWidget {
+  PeopleListScreen({Key key}) : super(key: key);
+
+  @override
+  _PeopleListScreenState createState() => _PeopleListScreenState();
+}
+
+class _PeopleListScreenState extends State<PeopleListScreen> {
+  List widgets = [];
```
Comment (lenkrzysztof) [inline lib/people/people_list_screen.dart]: It should be `posts` list

Comment (lenkrzysztof) [reply lib/people/people_list_screen.dart]: If you create a model and add here as a generic like `List<Post>` flutter will infer this type in bellowed code.

File: lib/people/people_list_screen.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  void initState() {
+    super.initState();
+    loadData();
+  }
+
+  Future<void> loadData() async {
+    var dataURL = Uri.parse('https://jsonplaceholder.typicode.com/posts');
+    http.Response response = await http.get(dataURL);
+    setState(() {
+      widgets = jsonDecode(response.body);
```
Comment (lenkrzysztof) [inline lib/people/people_list_screen.dart]: Remember to use `utf8.decode(response.bodyBytes);` if your response can include different chars.

Comment (lenkrzysztof) [reply lib/people/people_list_screen.dart]:
```
response.map<Post>(
              (site) => Post.fromJson(site as Map<String, dynamic>))
          .toList() as List<Post>;
```

Comment (lenkrzysztof) [reply lib/people/people_list_screen.dart]: This is how we handle async API calls:
```
try {
      final response = await ...
      final decodeData = utf8.decode(response.bodyBytes);
      final list = jsonDecode(decodeData);

     list.map<Post>(
              (site) => Post.fromJson(site as Map<String, dynamic>))
          .toList() as List<Post>;
    } on ApiError catch (error) {
      final logger = Logger();
      logger.e('ApiError in method', error);
    } on SocketException {
      throw ConnectionException();
    } catch (error, stack) {
      final logger = Logger();
      logger.e(error);
      throw UnknownException();
    }
```

File: lib/piechart/my_slider.dart [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
```
Comment (lenkrzysztof) [inline lib/piechart/my_slider.dart:1]: No need to import both cupertino and material. I always use material import.

File: lib/piechart/piechart_home_screen.dart [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import 'my_chart.dart';
+import 'my_slider.dart';
+
+class PiechartHomeScreen extends StatefulWidget {
+  @override
+  _PiechartHomeScreenState createState() => _PiechartHomeScreenState();
+}
+
+class _PiechartHomeScreenState extends State<PiechartHomeScreen> {
```
Comment (lenkrzysztof) [inline lib/piechart/piechart_home_screen.dart]: If you not use state no need to create stateful widgets.

**Comment by Swirastlynn** [comment]
> Which version of Flutter do you use? If 2.5 your constants value should have const keyword before like: `const Text('example');`. You should checked your terminal due to warning then.

Ha! `1.2.0`, so I `flutter upgrade`

File: lib/people/people_list_screen.dart (hunk 3 above)
Comment (Swirastlynn) [reply lib/people/people_list_screen.dart]: > ```
> response.map<Post>(
>               (site) => Post.fromJson(site as Map<String, dynamic>))
>           .toList() as List<Post>;
> ```

Why `site` naming?

File: lib/general/network/api_manager.dart [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    try {
+      final httpResponse = await http.get(url);
+      jsonResponse = _validateResponse(httpResponse);
+    } on SocketException {
+      throw NoInternetConnectionException();
+    }
+    return jsonResponse;
+  }
+
+  dynamic _validateResponse(http.Response response) async {
```
Comment (KrzysztofLen) [inline lib/general/network/api_manager.dart]: no need to be async since there is no await keyword

Comment (Swirastlynn) [reply lib/general/network/api_manager.dart]: Updated here https://github.com/Swirastlynn/skitour-planning/tree/bloc-architecture

**Pull request merged**